### DISCUSSION TITLE
careful

### DISCUSSION THREAD

**Discussion by sgamesdev** [Leo-Corporation/Gavilya#277] [Show and tell]
you should find a solution to hide the RAWGAPIKey , your  api key is ending in "e25a6b18"  , if in the future you will want to implement a database , you should have a solution

**Comment by lpeyr** [comment]
Ok, thanks. Will try to address this issue, but what do you mean by "implementing a database"?

**Reply by sgamesdev** [reply]
i am just saying that if you have other projects that works with databases , you should hide the connection string someway

**Reply by lpeyr** [reply]
Ok, thanks for your help, I will try to fix it :)

**Comment by sgamesdev** [comment]
I found a way but it has a downside , basically you put the api key in app settings and encrypt it , the downside is that the encryption key is stored locally so it won't work on other computers

**Reply by lpeyr** [reply]
Thanks for the suggestion, but I think I will protect the key by doing this:
- Instead of storing it as plaintext, I will encrypt it using AES encryption, and I'll store the encrypted string.
- The encryption key will be stored as an encrypted `byte[]` array that can only be decrypted using the `ProtectData` class of `System.Security.Cryptography`.
- Obviously, the encryption system won't be published publicly, but I'll leave some security advices in the public `APIKeys` class.

I know you could still retrieve the API key by doing some reverse engineering or by intercepting the request sent to the API, but this is still better than just storing it in plaintext.
